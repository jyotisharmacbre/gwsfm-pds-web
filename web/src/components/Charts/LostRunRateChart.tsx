import React from 'react';

import { Doughnut } from 'react-chartjs-2';

const dataLost = {
  datasets: [
    {
      data: [4, 2, 7, 9, 23, 2],
      backgroundColor: [
        // '#ced0fe',
        // '#7e83fe','#704ba4', '#4cbd7f',
        // '#f3ca55'
        '#CEF1A3',
        '#526061',
        '#FBD55B',
        '#935FD6',
        '#EA8032',
        '#3DAAE9'
      ],
      label: 'My dataset' // for legend
    }
  ],
  labels: [
    'Initial Customer Enquiry',
    'J&A',
    'Bid Submitted',
    'Order Received',
    'In Progress',
    'Completed'
    //ICE, J&A, Bid Submitted, Order Received, In Progress and Completed
  ]
};

const optionshideLabels = {
  showTotalLabel: true,
  cutoutPercentage: 70,
  legend: {
    display: false
  } //, maintainAspectRatio:false
};

export default function LostRunRateChart(props: {
  ICE: number;
  JandA: number;
  BidSubmitted: number;
  OrderReceived: number;
  InProgress: number;
  Completed: number;
}) {
  return <Doughnut data={dataLost} options={optionshideLabels} />;
}
