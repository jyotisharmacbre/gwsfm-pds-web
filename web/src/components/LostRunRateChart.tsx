import React from 'react';

import { Doughnut, Pie, Polar, Bubble, Line } from 'react-chartjs-2';
import Grid from '@material-ui/core/Grid';

  const dataLost = {
    datasets: [{
      data: [
        4,2,7,9,23,2
        
      ],
      backgroundColor: [
        // '#ced0fe',
        // '#7e83fe','#704ba4', '#4cbd7f',
        // '#f3ca55'
        '#CEF1A3',
        '#526061','#FBD55B', '#935FD6',
        '#EA8032','#3DAAE9'
       
      ],
      label: 'My dataset' // for legend
    }],
    labels: [
      'Initial Customer Enquiry',
      'J&A',
      'Bid Submitted','Order Received','In Progress','Completed'
      //ICE, J&A, Bid Submitted, Order Received, In Progress and Completed
    ]
  };
 
  const optionshideLabels={  
 
      legend: {
        display: false
      }//, maintainAspectRatio:false     
};

export default function LostRunRateChart(props: {ICE:number, JandA:number, BidSubmitted:number, OrderReceived:number, InProgress:number , Completed:number}){
    return (  <Doughnut   data={dataLost} options={optionshideLabels} /> 
             );
   
   
  }
  