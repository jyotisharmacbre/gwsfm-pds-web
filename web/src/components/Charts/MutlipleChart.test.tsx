import React from 'react';
import ReactDOM from 'react-dom';
import LostRunRateChart from './LostRunRateChart';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MultipleChart from './MultipleCharts';
Enzyme.configure({ adapter: new Adapter() });

it('Lost Run Rate & OnHold Chart should render', () => {
    const component = shallow(
      <MultipleChart ProjectTotal={200} ICE={50} Rejected={50} JandA={50} LostProjects={30} OrderReceived={20} OnHoldProject={15} InProgress={24} Completed={20}/> 
      );
     // let tree = component.toJSON();
     expect(component.getElements()).toMatchSnapshot();
});

 it('Graph Grid exist', () => {
   const props ={
   ProjectTotal:200 , ICE:56, JandA:12, Rejected:12,LostProjects:40, OrderReceived:34, InProgress:56, Completed:45,OnHoldProject:30
   }
     const component = mount(
        <MultipleChart {...props} />
    );

    expect(component.find('#GridMultipleChart').exists);  
 });

 
