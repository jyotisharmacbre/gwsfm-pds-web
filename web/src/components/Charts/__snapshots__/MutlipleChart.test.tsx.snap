// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lost Run Rate & OnHold Chart should render 1`] = `
Array [
  <ForwardRef(WithStyles)
    container={true}
    id="GridMultipleChart"
    min-Height={350}
    spacing={0}
  >
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        Initial Enquiry
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#CEF1A3",
                  ],
                  "data": Array [
                    50,
                    150,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "Initial Customer Enquiry",
                "Other Projects",
              ],
            }
          }
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
          type={
            Object {
              "originalDoughnutDraw": [Function],
            }
          }
        />
      </div>
       
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        Rejected
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
         
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#FBD55B",
                  ],
                  "data": Array [
                    50,
                    150,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "Rejected",
                "Other Projects",
              ],
            }
          }
          height={300}
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
        />
      </div>
       
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        J&A
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#526061",
                  ],
                  "data": Array [
                    50,
                    150,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "J&A",
                "Other Projects",
              ],
            }
          }
          height={300}
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
        />
      </div>
       
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        Order Received
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#935FD6",
                  ],
                  "data": Array [
                    20,
                    180,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "Order Received",
                "Other Projects",
              ],
            }
          }
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
        />
      </div>
       
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        In Progress
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#EA8032",
                  ],
                  "data": Array [
                    24,
                    176,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "In Progress",
                "Other Projects",
              ],
            }
          }
          height={300}
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
        />
         
      </div>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={2}
      md={2}
      sm={6}
      style={
        Object {
          "height": "280px",
          "textAlign": "center",
        }
      }
      xs={6}
    >
      <h4>
        Completed
      </h4>
      <div
        style={
          Object {
            "height": "250px",
            "textAlign": "center",
          }
        }
      >
        <Doughnut
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#3DAAE9",
                  ],
                  "data": Array [
                    20,
                    180,
                  ],
                  "label": "My dataset",
                },
              ],
              "labels": Array [
                "J&A",
                "Other Projects",
              ],
            }
          }
          height={300}
          options={
            Object {
              "cutoutPercentage": 75,
              "elements": Object {
                "center": Object {
                  "color": "#FF6384",
                  "fontStyle": "Arial",
                  "sidePadding": 20,
                  "text": "90%",
                },
              },
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "showfirstItemLabel": true,
            }
          }
        />
      </div>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      lg={12}
      md={12}
      sm={12}
      style={
        Object {
          "padding": "10px",
          "textAlign": "center",
        }
      }
      xs={12}
    >
       
      <h3>
        Total : 
        200
         Lost : 
        30
         On-Hold :
         
        15
         
      </h3>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>,
]
`;
